From: Ignace Mouzannar <mouzannar@gmail.com>
Subject: Corrects spelling errors (seperated -> separated) in README file
 Corrects minor typo in src/wiki.c

diff -urNad didiwiki-0.5~/README didiwiki-0.5/README
--- didiwiki-0.5~/README	2009-10-06 23:12:56.000000000 +0200
+++ didiwiki-0.5/README	2009-10-06 23:14:06.000000000 +0200
@@ -45,7 +45,7 @@
 
 http://didiwiki/api/pages
 
- Returns a seperated list ( one entry per line containing page
+ Returns a separated list ( one entry per line containing page
  title,TAB,modified date) of wiki pages with the most recently modified
  first.
 
diff -urNad didiwiki-0.5~/src/wiki.c didiwiki-0.5/src/wiki.c
--- didiwiki-0.5~/src/wiki.c	2009-10-06 23:14:06.000000000 +0200
+++ didiwiki-0.5/src/wiki.c	2009-10-06 23:15:58.000000000 +0200
@@ -730,7 +730,7 @@
   if (expr == NULL || strlen(expr) == 0)
     {
       wiki_show_header(res, "Search", FALSE);
-      http_response_printf(res, "No Search Terms supplied");
+      http_response_printf(res, "No search terms supplied.");
       wiki_show_footer(res);
       http_response_send(res);
       exit(0);
diff -urNad didiwiki-0.5~/src/wikitext.h didiwiki-0.5/src/wikitext.h
--- didiwiki-0.5~/src/wikitext.h	2009-10-06 23:14:06.000000000 +0200
+++ didiwiki-0.5/src/wikitext.h	2009-10-06 23:14:06.000000000 +0200
@@ -108,7 +108,7 @@
 " ---- Horizonal line\n"                                                 \
 "----\n"                                                                 \
 "\n"                                                                     \
-"Paragraphs are seperated by an empty line\n"                            \
+"Paragraphs are separated by an empty line\n"                            \
 "\n"                                                                     \
 "Like this. Another paragraph.\n"                                        \
 "\n"                                                                     \
